src/js/util.js

```diff
@@ -250,6 +250,33 @@ function deprecate(fn, msg, code) {
   return deprecated;
 }
 
+var customPromisifySymbol = 'util:promisify:custom';
+promisify.custom = customPromisifySymbol;
+function promisify(original) {
+  if (typeof original != 'function') {
+    throw new TypeError('expect a function on promisify');
+  }
+  if (typeof original[customPromisifySymbol] == 'function') {
+    return original[customPromisifySymbol];
+  }
+  return function promisified() {
+    var args = Array.prototype.slice.call(arguments, 0);
+
+    return new Promise((resolve, reject) => {
+      args.push(callback);
+      original.apply(this, args);
+
+      function callback(err, result) {
+        if (err != null) {
```

Comment:
Should we verify this is an `Error` instance?

Reply:
No

Reply:
I saw the same function on Node.js looks like this:

```js
if (err)
  return reject(err)
```

Why do we make this difference here?

Reply:
`if (err)` would be true if err is truthy, `err != null` can only be true if err is `null` or `undefined`.

Reply:
Yay! How about fixing it in Node.js core, either?